pcsx2/IPC.cpp

```diff
@@ -76,17 +76,22 @@ SocketIPC::SocketIPC(SysCoreThread* vm, unsigned int slot)
 	}
 
 #else
+	char* runtime_dir = nullptr;
```

Comment:
How about using an std::string for all this?

Reply:
We deal here with the C API all over, both from sockets and getenv.
Using std::string would only add unneeded overhead and conversions everywhere.